### PR TITLE
[Longformer] Multiple choice for longformer

### PR DESCRIPTION
**Pull request by patrickvonplaten**
## Description:

- This PR adds Multiple Choice for Longformer as in: #4644 (Sorry for not telling you earlier @patil-suraj). @ibeltagy 

- The documentation is updated for all models using MultipleChoice since their `input_ids` have 3 dimensions. It is done by use of `{}.format()` for `INPUTS_DOCSTRING`. @LysandreJik 

- Adds a couple of models that were missing to the respective `models_page` @LysandreJik 

@ibeltagy Regarding global attention - I think we probably should automatically add global attention here since the multiple inputs are flattened across the dimension `num_choices` and then should attend each other. Maybe add global attention always on the first token of each input of the dim `num_choices`?

- Global attention should probably still be implemented. Waiting for @ibeltagy answer.

### CODE REVIEW

**Comment by patil-suraj** [comment]
As mention in the paper, they used global  attention on all answer candidates for WikiHop, so maybe we can use global attention on all choice tokens ?

**Comment by ibeltagy** [comment]
Looks like multiple choice models are used for datasets with short input like swag (question + multiple choices), and datasets with long context like Wikihop (question + multiple choices + a long context). For the second use case, as @patil-suraj mentioned, we need global attention on all tokens of the question and the choices. 

For the first use case, I am not sure how to handle it. Maybe global attention everywhere, but in this case, it is equivalent to n^2 attention. Do you know of a dataset of the first use case with longer inputs than swag?

**Comment by patil-suraj** [comment]
@ibeltagy If we have such multiple use-cases then I think it would be better if we leave this to the user. Not entirely sure though.

**Comment by ibeltagy** [comment]
@patil-suraj, we can leave it to the user, or we can just do as you suggested earlier, put global attention on the question and all choices.

File: src/transformers/modeling_longformer.py [hunk 1]
```diff
@@ -1070,3 +1070,100 @@ def forward(
...
+
+            Attentions weights after the attention softmax, used to compute the weighted average in the self-attention
+            heads.
+
+    Examples::
+
+        from transformers import LongformerTokenizer, LongformerForTokenClassification
+        import torch
+
+        tokenizer = LongformerTokenizer.from_pretrained('longformer-base-4096')
```
Comment (ibeltagy) [inline src/transformers/modeling_longformer.py:1135]: Model name was changed to `allenai/longformer-base-4096`.

**Comment by patrickvonplaten** [comment]
Yeah I think ideally we would actually leave it to the user with a `global_attention` input argument maybe. We could actually have this for all forward functions...I'll think a bit about it tomorrow!

**Comment by patrickvonplaten** [comment]
We had some internal discussion and decided to change / extend the API of `Longformer` slightly. 

We will have to input tensors `attention_mask` (as usual composed of 0's and 1's only) and a `global_attention_mask` also composed of zeros and ones (0 => local attention; 1=> global attention). If `global_attention` is not defined by the user we create if necessary `LongformerForQuestionAnswering` and maybe `LongformerForMultipleChoice`. Is that ok for you @ibeltagy ? 

We will keep the inner workings the same (merge `attention_mask` with `global_attention_mask`) but make sure the user has a more intuitive API, since people always think of masks as boolean tensors in this library. 

I will merge this PR and open a new one with the proposed changes.
